mdata/cache/ccache_metric.go

```diff
@@ -345,16 +345,22 @@ func (mc *CCacheMetric) searchForward(ctx context.Context, metric schema.AMKey,
 	// add all consecutive chunks to search results, starting at the one containing "from"
 	for ; ts != 0; ts = mc.chunks[ts].Next {
 		log.Debugf("CCacheMetric searchForward: forward search adds chunk ts %d to start", ts)
-		res.Start = append(res.Start, mc.chunks[ts].Itgen)
+		chunk, ok := mc.chunks[ts]
+		if !ok {
+			log.Errorf("CCacheMetric searchForward: chunk with ts %d is not cached", ts)
+			res.Type = Miss
+			return
```

Comment:
i'd `break` instead of `return`.... because before we got here some chunks might have already been added to `res.Start`, in which case we'd want `res.Type` to be set to `HitPartial` on `:372`